### DISCUSSION TITLE
[phpunit-bridge] How to lock dependencies

### DISCUSSION THREAD

**Discussion by lazka** [symfony/symfony#44961] [Q&A]
Hey,

last week some transitive dependency update of phpunit conflicted with my code under test and broke my build. This also affected older releases since the phpunit pulled via phpunit-bridge isn't pinned and was loading the broken version there as well, so a rollback didn't help.

I've in the meantime switched to requiring phpunit directly, even if the docs don't recommend it.

I'm wondering how everyone else is handling this?
How can I pin php package versions used during testing with phpunit-bridge?

**Comment by GromNaN** [comment]
There is the `SYMFONY_PHPUNIT_REQUIRE` env var that can be used to add a package, don't konw if it can be used to lock a package to a specific version.

**Reply by lazka** [reply]
Looking at the code it should be possible to pass a version specifier, but the problem is I don't know the transitive dependencies of phpunit, so that's not very practical.

**Reply by ghost** [reply]
I don't quite understand. If you set SYMFONY_PHPUNIT_REQUIRE=${version} then phpunit-bridge will always install the same phpunit and it's dependencies.

Unless phpunit has dependencies that do not follow semantic versioning i.e introduce BC changes in minor versions. Then this is a problem when using phpunit-bridge indeed.

**Reply by lazka** [reply]
Not sure. I have "phpdocumentor/type-resolver" 1.5.1 in the project, and phpunit-bridge installed 1.6.0 as a transitive dependency, which results in type-resolver erroring out during tests. Upgrading my project dep from 1.5.1 to 1.6.0 too made it work again.

So somehow the having two different versions of "type-resolver" fails.

I don't think this can be solved by adhering to semver.
